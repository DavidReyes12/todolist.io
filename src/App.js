import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom';

import ProtectedRoute from './permissions/ProtectedRoute';
import AuthRoute from './permissions/AuthRoute';
import Home from "./views/Home"
import Auth from './views/Auth';

import './App.css';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>

          <Route exact path='/home' element={<ProtectedRoute />}>
            <Route path='/home/*' element={ <Home />} />
            <Route path="/home" element={ <Navigate to="/home/todolist" replace /> } />
          </Route>

          <Route exact path='/auth' element={<AuthRoute />}>
            <Route path='/auth/*' element={ <Auth />} />
            <Route path="/auth" element={ <Navigate to="/auth/Login" replace /> } />
          </Route>

          <Route path="/*" element={ <Navigate to="/auth" replace /> } />

        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
