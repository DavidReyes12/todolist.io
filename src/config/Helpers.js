export const loadFromLocalStorage = () => {
    try {
      const appState = localStorage.getItem("appSession");
      if (!appState) return null;
      return JSON.parse(appState);
    } catch (e) {
      console.log(e);
      return null;
    }
};
  
export const saveToLocalStorage = (state) => {
    try {
        const appState = JSON.stringify(state);
        localStorage.setItem("appSession", appState);
    } catch (e) {
        console.log(e);
    }
};