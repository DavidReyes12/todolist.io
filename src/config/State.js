import React, { useState, useEffect } from 'react'
import { loadFromLocalStorage, saveToLocalStorage } from "./Helpers";

export const SessionContext = React.createContext(null);

export const State = ({ children }) => {

    const [session, setSession] = useState(loadFromLocalStorage());

    useEffect(() => {
        saveToLocalStorage(session);
    }, [session]);
    

    return (
        <SessionContext.Provider value={{ session, setSession }}>
            { children }
        </SessionContext.Provider>
    );
};

