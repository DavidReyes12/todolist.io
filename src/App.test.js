import renderer from 'react-test-renderer';
import App from './App';
import { SessionContext } from "./config/State";

const session = {
  userName: "David Reyes",
  id: 1,
  welcomeMsg: ""
};

const setSession = () => {};

describe('App test', () => {
  test('App test with Snapshot', () =>{
      const tree = renderer.create(
        <SessionContext.Provider value={{session, setSession}}>
          <App />          
        </SessionContext.Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
  })
})
