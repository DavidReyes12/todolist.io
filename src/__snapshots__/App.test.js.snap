// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test App test with Snapshot 1`] = `
<div
  className="App"
/>
`;
